### DISCUSSION TITLE
Jop shop problem with larger horizon runs so slow and takes up so many Cpu rate.

### DISCUSSION THREAD

**Discussion by Bestbbb** [google/or-tools#3514] [CP-SAT questions] [answered]
Version: v9.4/
Language: Java
Solver:CP-SAT 
System:windows
Windows

The example in https://developers.google.com/optimization/scheduling/job_shop  was done at the granularity of days,in my experiment,I program at the minute level,the amount of horizon is about 24000 minutes,and then the problem is that ,Cpu rates almost can be up to 97.9,my cpu is intel i5 11400F,I change another machine which is Intel Xeon Platinum 8260L,both of them can not make the program run in a suitable time.I wait about 14 hours and can see the result.
I want to make it into a real factory experience,so I think the efficiency and time value a lot,Could you please tell me how can I improve the speed of this situation?By the way,if I use the normal day level,result will be fine.
I guess that may be caused by big number.
If you can help me ,I will be very grateful.(#^.^#)

**Comment by lperron** [comment, marked as answer]
1) if you do not specify it, the solver uses all cores
2) finding a solution is another issue. Do you have a log enabled? Have you set a time limit?

**Reply by Bestbbb** [reply]
Sorry,I did not set the time limit and log enabled,how can I search the related api of time limit in JAVA.Thank you

**Reply by lperron** [reply]
https://developers.google.com/optimization/cp/cp_tasks#time-limit

**Reply by Bestbbb** [reply]
Thanks a lot.

**Reply by lperron** [reply]
number of workers: parameter num_search_workers or num_workers
log: log_search_progress (set to true)

**Reply by Bestbbb** [reply]
Another question,if I limit the time,Is it possible that the solution is not optimal？

**Reply by lperron** [reply]
of course, it is possible. These are NP-Hard problems. If the solver does not say OPTIMAL, you can not know.

**Reply by Bestbbb** [reply]
Got it,Thank you for your patience

**Reply by Bestbbb** [reply]
I was wondering if there would be different efficiencies with different programming languages, maybe c++ is faster than both python and java?

**Comment by lperron** [comment]
Solving is in C++.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00
